pkg/apis/v1alpha2/crd_validation_cel_test.go

```diff
@@ -151,6 +151,7 @@ var _ = Describe("CEL/Validation", func() {
 				},
 			}
 			Expect(env.Client.Create(ctx, nodeClass)).To(Succeed())
+			Expect(env.Client.Delete(ctx, nodeClass)).To(Succeed())
```

Comment:
nit: Deleting all AKSNodeClass instances in AfterEach instead would be better (and would fix leaks in other tests - however inconsequential they are, due to random names ...)